src/tfr_executive/src/autonomous_action_server.cpp

```diff
@@ -80,39 +124,189 @@ class AutonomousExecutive
          * ACTION COMPONENTS
          * - Goal: none
          * - Feedback: none
-         * - Result: none
-         */ 
+         * - Result: none */ 
         void autonomousMission(const tfr_msgs::EmptyGoalConstPtr &goal)
         {
+            
             ROS_INFO("Autonomous Action Server: mission started");
+            if (server.isPreemptRequested() || ! ros::ok())
+            {
+                server.setPreempted();
+                return;
+            }
+
+            if (LOCALIZATION)
+            {
+                ROS_INFO("Autonomous Action Server: commencing localization");
+                tfr_msgs::EmptyGoal goal{};
+                localizationClient.sendGoal(goal);
+                //handle preemption
+                while (!localizationClient.getState().isDone())
+                {
+                    if (server.isPreemptRequested() || ! ros::ok())
+                    {
+                        localizationClient.cancelAllGoals();
+                        server.setPreempted();
+                        ROS_INFO("Autonomous Action Server: localization preempted");
+                        return;
+                    }
+                    frequency.sleep();
+                }
+                if (localizationClient.getState()!=actionlib::SimpleClientGoalState::SUCCEEDED)
+                {
+                    ROS_INFO("Autonomous Action Server: localization failed");
+                    server.setAborted();
+                    return;
+                }
+                ROS_INFO("Autonomous Action Server: localization finished");
+            }
+
+            if (NAVIGATION_TO)
+            {
+                ROS_INFO("Autonomous Action Server: commencing navigation");
+ 
+                tfr_msgs::NavigationGoal goal;
+                //messages can't support user defined types
+                goal.location_code= static_cast<uint8_t>(tfr_utilities::LocationCode::MINING);
+                navigationClient.sendGoal(goal);
```

Comment:
Can you describe some of the data flow for me here?  Navigation needs to know the point where we're going, and I thought that was calculated during Localization.  If Localization isn't returning that (or maybe more accurately, returning the robot's relative position), how is that information getting to Navigation?

Reply:
Happy to.
_Localization_ Here we figure out from the cameras where we are. We take that information and send it to a node called the `bin_broadcaster`, it's job is to repeatedly broadcast transforms of where the bin is in the `tf2` transform system, which is integrated with our odometry and sensors. From the moment we can translate any point, from any reference frame in the system and find it's distance from the bin.

_Navigation_ We know we to travel 5.2 m east of the bin. Here is how that works in code The navigation stack takes an xyz quaternion command, in relation to a `tf2` reference frame. All we have to do  is tell it to go +5.2m in the "east" direction, and say that these measurements are in relation to the bin reference frame, and it will take us to the correct place.

Reply:
However what I have just described happens at the navigation action server, and localization action server level. Up here at executive we don't have as much detail.

Reply:
So the transform publisher is set up during Localization and continues to execute during Navigation?